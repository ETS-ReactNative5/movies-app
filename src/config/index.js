export const api_key = '8c793d948d52df975cb1f74b7c39c00f';