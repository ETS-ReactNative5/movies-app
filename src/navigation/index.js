export { Navigation } from './main-stack';