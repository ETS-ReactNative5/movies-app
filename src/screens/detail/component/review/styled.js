import FastImage from "react-native-fast-image";
import styled from "styled-components/native"


export const Container = styled.View`
    flex:1;
    flex-direction: row;
    align-items: center;
    padding-vertical: 20px;
    padding-right: 12px;
`;

export const ImageWrapper = styled.View`
    margin: 8px;
    align-self: flex-start;
`;

export const Image = styled(FastImage)`
    resize-mode: cover;
    width: 80px;
    height: 80px;  
    border-radius: 100px;
`;

export const MovieReviewWrapper = styled.View`
    flex:1;
`;
export const UserName = styled.Text`
    font-weight: bold;
    width: 75%;
`;
export const UserInfo = styled.View`
`;
export const AverageScore = styled.View`
    border-radius: 5px;
    background: black;
    opacity: 0.6;
    padding-horizontal: 18px
    padding-vertical: 6px;
`;
export const ReviewData = styled.Text`
    margin-vertical: 5px;
`;

export const ReviewText = styled.Text``;

export const StyledWrapper = styled.View`
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
`;
