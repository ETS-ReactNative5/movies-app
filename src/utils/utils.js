
export const getTruncatedString = (str, num) => {
  if (str?.length <= num || num === -1) {
    return str;
  }
  return `${str?.slice(0, num)}...`;
};

export const bannerImageSrc = 'https://images.unsplash.com/photo-1517458776179-baae60e41ef2?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=3270&q=80';
