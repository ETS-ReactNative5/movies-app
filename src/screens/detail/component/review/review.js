
import React from 'react';
import { Text } from 'react-native';

import FastImage from 'react-native-fast-image';
import { emptyPlaceholder, getTruncatedString, imagePrefixUrl, TEST_IDS } from '../../../../utils/index';

import {
    ReviewText,
    ReviewData,
    AverageScore,
    UserInfo,
    UserName,
    Image,
    MovieReviewWrapper,
    ImageWrapper,
    Container,
    StyledWrapper,
} from './styled';

export const ReviewCard = ({ review }) => {

    const {
        content,
        created_at,
        author_details
    } = review;

    const { name, rating, avatar_path } = author_details;
    const revieweeName = name ? `A review by ${name}` : 'Anonymous';

    let options = { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' };
    const formattedDate = new Date(created_at).toLocaleString('en-US', options);
    
    const getImageUrl = (url) => {
        if (url?.includes('https')) {
            return url.slice(1);
        } else if (url) {
            return imagePrefixUrl + url;
        } else {
            return emptyPlaceholder;
        }
    }

    return (
        <Container>
            <ImageWrapper>
                <Image
                    source={{ uri: getImageUrl(avatar_path) }}
                    resizeMode={FastImage.resizeMode.contain}
                />
            </ImageWrapper>

            <MovieReviewWrapper>
                <UserInfo>
                    <StyledWrapper>
                        <UserName testID={TEST_IDS.REVIEW_CARD_REVIEWEE_NAME} >
                            {revieweeName}
                        </UserName>
                        {
                            rating && (
                                <AverageScore >
                                    <Text
                                        style={{ color: 'white' }}
                                        testID={TEST_IDS.REVIEW_CARD_AVERAGE_SCORE}
                                    >
                                        {rating}
                                    </Text>
                                </AverageScore>)
                        }

                    </StyledWrapper>
                    {
                        formattedDate && (
                            <ReviewData testID={TEST_IDS.REVIEW_CARD_DATE} >
                                {`Written on ${formattedDate}`}
                            </ReviewData>
                        )
                    }

                </UserInfo>

                <ReviewText testID={TEST_IDS.REVIEW_CARD_CONTENT} >
                    {getTruncatedString(content, 200)}
                </ReviewText>

            </MovieReviewWrapper>

        </Container>

    )
};

