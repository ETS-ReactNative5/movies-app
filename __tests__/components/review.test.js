import React from 'react';
import { render } from '@testing-library/react-native';
import { TEST_IDS } from '../../src/utils';

import { ReviewCard } from '../../src/screens/detail/component/review';

const props = {
    "content": "As I'm writing this review",
    "author": "Cat Ellington",
    "author_details": {
        "name": "Cat",
        "username": "CatEllington",
        "avatar_path": "/uCmwgSbJAcHqNwSvQvTv2dB95tx.jpg",
        "rating": 8
    },
    "created_at": "2017-02-13T22:23:01.268Z",
};

describe('Review Card Component of Detail Screen', () => {

    it('should match the Review Card component snapshot', () => {
        const { toJSON } = render(
            <ReviewCard
                review={props}
            />
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('should render name of reviewee correctly', async () => {
        const { getByTestId } = render(
            <ReviewCard
                review={props}
            />);
        expect(getByTestId(TEST_IDS.REVIEW_CARD_REVIEWEE_NAME).props.children).toBe('A review by Cat');
    });

    it('should render movie rating correctly', async () => {
        const { getByTestId } = render(
            <ReviewCard
                review={props}
            />);
        expect(getByTestId(TEST_IDS.REVIEW_CARD_AVERAGE_SCORE).props.children).toBe(8);
    });

    it('should render movie comment correctly', async () => {
        const { getByTestId } = render(
            <ReviewCard
                review={props}
            />);
        expect(getByTestId(TEST_IDS.REVIEW_CARD_CONTENT).props.children).toBe(`As I'm writing this review`);
    });

    it('should render movie comment date correctly', async () => {
        const { getByTestId } = render(
            <ReviewCard
                review={props}
            />);
        expect(getByTestId(TEST_IDS.REVIEW_CARD_DATE).props.children).toBe(`Written on Tuesday, February 14, 2017`);
    });

});
