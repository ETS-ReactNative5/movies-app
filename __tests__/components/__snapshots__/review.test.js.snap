// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Card Component of Detail Screen should match the Review Card component snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingRight": 12,
        "paddingVertical": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "marginBottom": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 8,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderRadius": 100,
              "height": 80,
              "resizeMode": "cover",
              "width": 80,
            },
          ],
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://image.tmdb.org/t/p/w500//uCmwgSbJAcHqNwSvQvTv2dB95tx.jpg",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontWeight": "bold",
                "width": "75%",
              },
            ]
          }
          testID="reviewee_name"
        >
          A review by Cat
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "black",
                "borderRadius": 5,
                "opacity": 0.6,
                "paddingHorizontal": 18,
                "paddingVertical": 6,
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
            testID="average-score"
          >
            8
          </Text>
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "marginVertical": 5,
            },
          ]
        }
        testID="data"
      >
        Written on Tuesday, February 14, 2017
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {},
        ]
      }
      testID="content"
    >
      As I'm writing this review
    </Text>
  </View>
</View>
`;
